### PR TITLE
Adding my broken yaml for testing + Added pre-commit hook for poetry lock

### PR DESCRIPTION
**Pull request by ctr26**
This can be integrated as a test in pytest similar to

https://github.com/BioImage-Archive/bia-agent/blob/main/bia_agent/tests/test_cli.py

Probably best as a seperate test which finds all yawls in tests and uses that as a pytest fixture

### CODE REVIEW

**Comment by ctr26** [comment]
@sherwoodf 

Do you have any smart ideas on how to test for bad yamls like this which produce invalid pagetab?

My feeling is it needs a pagetab validator (work)

**Comment by sherwoodf** [comment]
So there's:
Invalid yaml - I think Liviu dealt with those by improving error messaging from the parser.
Valid yaml that produces invalid pagetab - since the conversion should be deterministic i don't think it matters whether we validate the yaml or the pagetab (unless we can get pagetab from other sources). Validating the yaml might be easier: e.g. using https://pypi.org/project/schema/ like in this example https://www.andrewvillazon.com/validate-yaml-python-schema/.

**Comment by ctr26** [comment]
Note this is the only way I know how to to a poetry pre commit hook. 

The documentation doesn't work for me/make sense

https://python-poetry.org/docs/pre-commit-hooks/

**Review by liviuba — APPROVED** [review]
LGTM!

File: tests/broken_specimens.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+study:
```
Comment (sherwoodf) [inline tests/broken_specimens.yaml:1]: Feels like the pre-commit hook should have been a separate PR to this - but i'm not against including bad yamls for future testing.

**Review by sherwoodf — APPROVED** [review]

File: tests/broken_specimens.yaml (hunk 1 above)
Comment (ctr26) [reply tests/broken_specimens.yaml:1]: Yeah, bad yaml is intentionally here,

**Pull request merged**